PYIC-5685: Remove local running setup from datastore

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Remove local running setup from datastore

### Why did it change

We had some logic to determine if core was running locally, and if so use a locally hosted version of dynamodb. This is a legacy thing that we don't use anymore, and can remove. Which will simplify a few things, including the config service.

Wynndow committed:
PYIC-5685: Remove local running setup from datastore

We had some logic to determine if core was running locally, and if so
use a locally hosted version of dynamodb. This is a legacy thing that we
don't use anymore, and can remove. Which will simplify a few things,
including the config service.

Joe-Edwards-GDS reviewed (commented):
Looks like this was used for local DynamoDB and SSM?

Interestingly, I was thinking about whether we want to move towards mock/local AWS dependencies - i.e. so we can run core-back entirely locally without an AWS account.
- DynamoDB using https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.html
- SSM using a stub that just reads from a JSON config file (this might also alleviate the need for complicated dev env overrides!)
- SQS using a stub that records audit events in memory and can be inspected by tests
- KMS using a stub that does encryption using the test keys

However, I'm not sure how much of the code you're removing will be useful for this, and having it affect all the unit tests feels like a bit of a red flag.

Once you sort the tests then this is probably fine...

Joe-Edwards-GDS commented on libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java:
```diff
@@ -77,7 +79,15 @@ public ConfigService(SSMProvider ssmProvider, SecretsProvider secretsProvider) {
+    @ExcludeFromGeneratedCoverageReport
+    public ConfigService() {
+        if (Boolean.parseBoolean(getEnvironmentVariable(UNMOCKABLE_CONFIG_SERVICE_TEST))) {
+            // For journey map tests that deserialize events that create a config service
```
Is this because the `BasicState` initialiser constructs a new config service each time?

This feels a bit suspect - I wonder if we can put the `configService` into `JourneyContext` instead, to avoid the states themselves being stateful (ha). If that makes this PR too complicated then we can leave it like this...

Wynndow replied:
...interesting. That'll be doable, and remove the need for the tests flag...

Wynndow committed:
BAU: Move ConfigService from BasicEvent to JourneyContext

This change means that BasicEvents can now be deserialized from the
journey maps without creating any ConfigService's. This was making it
hard to test, as we couldn't intercept and mock those ConfigServices.
They were trying to establish connections with AWS which was failing in
the pipelines.

By moving the ConfigService to the journey context, we can deserialize,
and mock, without issue.

Joe-Edwards-GDS reviewed (dismissed):
Love changes like this
![image](https://github.com/govuk-one-login/ipv-core-back/assets/142887793/581b255a-e0be-4841-85a9-672fce8b0c91)
